### PR TITLE
Update R CI logic

### PR DESCRIPTION
**Pull request by johnkerl**
Follow-on from https://github.com/single-cell-data/TileDB-SOMA/pull/285:

* The R code is now in an `apis/R` subdirectory; some spellings need updating
* Particular handling for Windows with `apis\R` vs `apis/R` when using YAML syntax
* Following advice from @eddelbuettel regarding use of helper scripts -- basic idea is to rely less on YAML syntax and more on shell-script syntax, as these are clear and unambiguously expressive

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
 
       - name: Install libgeos on Linux
         if: runner.os == 'Linux'
-        run: sudo apt-get update -y && sudo apt-get install -y libgeos-dev
+        run: sudo ./tools/ci/install-linux-dependencies.sh
```
Comment (eddelbuettel) [inline .github/workflows/R-CMD-check.yaml:37]: (nean == Not Even A Nit (TM))  I usually draw at 'too tiny script so I would have kept most shell expressions here, specifically one-liners but it is six of one and half a dozen of another and if you like it better the other way :man_shrugging:

File: tools/ci/build-and-check-r-package.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+echo "::group::Check TileDB R package"
+## set flag to tolerate missing suggested packages (as eg vignette building tools)
+export _R_CHECK_FORCE_SUGGESTS_=FALSE
+## set an 'under valgrind' variable (see SC-19185)
+export _RUNNING_UNDER_VALGRIND_=TRUE
+## tell valgrind to use '-s'
+export VALGRIND_OPTS="-s"
+## check package
+R CMD check --use-valgrind --as-cran --no-manual --ignore-vignettes tiledb_*.tar.gz
```
Comment (eddelbuettel) [inline tools/ci/build-and-check-r-package.sh:18]: Using valgrind is a nice touch but _expensive_. I doubt I would turn it on in each run.  You could make it an environment-variable toggle.  But again, if it works for you ...

File: tools/ci/install-pkgdown-dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+# Context: .github/workflows/*.yaml
+export R_REMOTES_NO_ERRORS_FROM_WARNINGS='true'
+Rscript -e 'install.packages("remotes")'
+Rscript -e 'install.packages("pkgdown")'
```
Comment (eddelbuettel) [inline tools/ci/install-pkgdown-dependencies.sh:5]: That is style wart:   `Rscript -e 'install.packages(c("remotes", "pkgdown"))'`  is more compact.

File: tools/ci/install-r-dependencies.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# Context: .github/workflows/*.yaml
+export R_REMOTES_NO_ERRORS_FROM_WARNINGS='true'
+Rscript -e 'install.packages("remotes")'
+Rscript -e 'install.packages("rcmdcheck")'
```
Comment (eddelbuettel) [inline tools/ci/install-r-dependencies.sh:5]: See above.

File: tools/ci/build-and-check-r-package.sh (hunk 2 above)
Comment (johnkerl) [reply tools/ci/build-and-check-r-package.sh:18]: ooooops this is a copy-paste error on my part, i did _not_ intend to use `valgrind` ... 👀

File: tools/ci/install-r-dependencies.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+# Context: .github/workflows/*.yaml
+export R_REMOTES_NO_ERRORS_FROM_WARNINGS='true'
+Rscript -e 'install.packages(c("remotes", "rcmdcheck"))'
```
Comment (eddelbuettel) [inline tools/ci/install-r-dependencies.sh:4]: You don't actually _use_ the `rcmdcheck` package as you invoke `R CMD check --as-cran --no-manual --ignore-vignettes tiledb_*.tar.gz`.  So you don't need to install it.

**Review by eddelbuettel — COMMENTED** [review]
Also:   R-CMD-check.yaml is buttugly if all other file in the directory are neatly lowercased.  Maybe r-pkg-check.yaml or something like that.  Another 'nean' ...

File: tools/ci/build-and-check-r-package.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+echo "::group::Build TileDB R package"
+## build package
+R CMD build --no-build-vignettes --no-manual .
+echo "::endgroup::"
+
+echo "::group::Check TileDB R package"
+## set flag to tolerate missing suggested packages (as e.g. vignette-building tools)
+export _R_CHECK_FORCE_SUGGESTS_=FALSE
+## check package
+R CMD check --as-cran --no-manual --ignore-vignettes tiledb_*.tar.gz
```
Comment (eddelbuettel) [inline tools/ci/build-and-check-r-package.sh:14]: That is a likely bug.  The package you copied this from is called `tiledb_X.Y.Z.tar.gz` so the file-globbing is correct.  Your package here appears to be `tiledbsoma` per the `Package: tiledbsoma` in `DESCRIPTION` so you need to take out the underscore there, or inject 'soma' in front of it.

**Review by eddelbuettel — CHANGES_REQUESTED** [review]

File: tools/ci/build-and-check-r-package.sh (hunk 6 above)
Comment (johnkerl) [reply tools/ci/build-and-check-r-package.sh:14]: https://github.com/single-cell-data/TileDB-SOMA/pull/296

thank you! :)

**Pull request merged**